sys/posix/select/posix_select.c

```diff
@@ -121,8 +121,10 @@ int select(int nfds, fd_set *readfds, fd_set *writefds, fd_set *errorfds,
             goto end;
         }
         if (!wait) {
-            /* timeout was reached */
-            break;
+            /* timeout was reached, timeout was not set, so no need to
+             * go to end */
+            errno = -EINTR;
```

Comment:
Shouldn't this and line 143 be setting `errno = EINTR`?

Reply:
Oops, yeah you are right. Too used to our API ^^"

Reply:
Fixed. Also removed an unnecessary `goto`